### PR TITLE
MC-36383: [B2B] A Product with Customizable Option (File) can not be added from a Requisition List to the Shopping Cart 

### PR DESCRIPTION
**Pull request by SmVladyslav**
### Related Pull Requests

magento/partners-magento2b2b/pull/503

### Fixed Issues (if relevant)

1. Fixes magento/partners-magento2b2b#377 (MC-36383)

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by SmVladyslav** [comment]
@magento run all tests

**Comment by rogyar** [comment]
For QA. 

The test coverage is present in the related B2B issue

**Comment by SmVladyslav** [comment]
@magento run Functional Tests EE

**Comment by rogyar** [comment]
Hi @SmVladyslav. Could you check the failing integration tests, please? They all related to the product options rendering. 

Thanks.

**Comment by SmVladyslav** [comment]
@magento run all tests

**Comment by SmVladyslav** [comment]
Hello, @rogyar! Integration tests have been fixed and now I am running builds again.

**Comment by SmVladyslav** [comment]
@magento run Functional Tests EE, WebAPI Tests

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B, Functional Tests EE

File: app/code/Magento/Catalog/Model/Product/Option/Type/File.php [hunk 1]
```diff
@@ -228,7 +228,10 @@ public function validateUserValue($values)
          * previous configuration with no newly uploaded file
          */
         $fileInfo = null;
-        if (isset($values[$option->getId()]) && is_array($values[$option->getId()])) {
+        if (isset($values[$option->getId()])) {
+            if (is_string($values[$option->getId()])) {
```
Comment (SmVladyslav) [inline app/code/Magento/Catalog/Model/Product/Option/Type/File.php:232]: The same approach is used at app/code/Magento/Catalog/Model/Product/Option/Type/Select.php:100

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B

**Review by zakdma — APPROVED** [review]

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario
**Preconditions:**
* Magento EE 2.4-develop installed
* B2B is installed
* Stores > Configuration > General > B2B Features > Enable Requisition List =Yes
* A Requisition List is created (for example, "Requisition List 1" : go to the storefront > My Account > My Requisition Lists > Create New Requisition List).
* A Simple Product is created with Customizable Option File (name = TEST*SIMPLE_FILE, SKU = TEST_SIMPLE*FILE, Customizable Options -> Option Title = Option Title = " OPTION 1", Option Type = File, Compatible File Extensions = jpg, Maximum Image Size = 1000 x 1000 px)

**Steps to Reproduce:**
* Go to the Storefront as the created Customer
* Go to the created Product page
* Select and upload a file
* Add the Product to the created Requisition List
* Go to the created Requisition List (My Account > My Requisition Lists )
* Check "Select all" checkbox
* Click the "Add to Cart" button

**Actual results:**

"You added 1 item(s) to your shopping cart." message appears
The Product is not added to the Shopping Cart
before


**AFTER** applying changes provided in the PR
* "You added 1 item(s) to your shopping cart." message appears
* The Product should be added to the Shopping Cart
after

**Comment by SmVladyslav** [comment]
@magento run Functional Tests B2B

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31093`.

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

The result is the same as in the report https://github.com/magento/magento2/pull/31093#issuecomment-737857774

**Pull request merged**